moml/core/__init__.py

```
@@ -29,30 +29,76 @@
+    
+    def find_charges_file(*args, **kwargs):
+        raise ImportError(f"find_charges_file requires additional dependencies: {e}")
+    
+    def read_charges_from_file(*args, **kwargs):
+        raise ImportError(f"read_charges_from_file requires additional dependencies: {e}")
 
 # Hierarchical graph coarsening
-from .hierarchical_graph_coarsener import GraphCoarsener
+try:
+    from .hierarchical_graph_coarsener import GraphCoarsener
+except ImportError as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)